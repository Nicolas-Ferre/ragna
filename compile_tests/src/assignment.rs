fn main() {}

#[ragna::gpu]
mod gpu {
    fn run() {
        let mut i32_value = 0;
        i32_value = 1.;
    }
}
