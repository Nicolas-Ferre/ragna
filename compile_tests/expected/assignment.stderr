error[E0308]: mismatched types
  --> compile_tests/src/assignment.rs:7:21
   |
7  |         i32_value = 1.;
   |                     ^^
   |                     |
   |                     expected integer, found floating-point number
   |                     arguments to this function are incorrect
   |
note: associated function defined here
  --> ./crates/ragna/src/values.rs:31:18
   |
31 |     pub const fn constant(value: T) -> Self
   |                  ^^^^^^^^
